ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -102,8 +102,52 @@ gcp_connection::delete_object(const std::string &object_key)
 
 // Retrieves an object from the google cloud bucket.
 int
-gcp_connection::get_object(const std::string &object_key, const std::string &path) const
+gcp_connection::read_object(const std::string &object_key, int64_t offset, size_t len, void *buf)
 {
+    bool exists = false;
+    size_t object_size;
+
+    // The object_exists function will check if the given object exists and print out any error
+    // messages.
+    if (object_exists(object_key, exists, object_size) != 0) {
+        return -1;
+    }
+
+    if (!exists) {
+        std::cerr << "Object '" << object_key << "' does not exist." << std::endl;
+        return -1;
+    }
+
+    if (offset < 0) {
+        std::cerr << "Offset " << offset << " is invalid. The offset cannot be less than zero."
+                  << std::endl;
+        return -1;
+    }
+
+    if (offset > object_size) {
```

Comment:
nit: `if (offset > object_size)` is a subset of `if (offset + len > object_size)` below. Do we need both?